packages/SwingSet/docs/run-policy.md

```diff
@@ -33,13 +33,16 @@ The kernel will invoke the following methods on the policy object (so all must e
 * `policy.vatCreated()`
 * `policy.crankComplete({ computrons })`
 * `policy.crankFailed()`
+* `policy.emptyCrank()`
 
 All methods should return `true` if the kernel should keep running, or `false` if it should stop.
 
 The `computrons` argument may be `undefined` (e.g. if the crank was delivered to a non-`xs worker`-based vat, such as the comms vat). The policy should probably treat this as equivalent to some "typical" number of computrons.
 
 `crankFailed` indicates the vat suffered an error during crank delivery, such as a metering fault, memory allocation fault, or fatal syscall. We do not currently have a way to measure the computron usage of failed cranks (many of the error cases are signaled by the worker process exiting with a distinctive status code, which does not give it an opportunity to report back detailed metering data). The run policy should assume the worst.
 
+`emptyCrank` indicates the kernel processed a queued messages which didn't result in a delivery.
```

Comment:
Half of me wants to not tell the runPolicy about these. I don't know how the host application author should react: how many are too many? I guess a basic approach would be to say "no more than 200 `emptyCrank`s in a block". Would that protect against some sort of infinite cyclic loop or DoS attack?

Let's expand on the comment a bit, add another sentence or two about what situation can cause this (`syscall.send` to a rejected promise?). It seems like anything pulling from a vat-outbound-queue will never result in a delivery, so maybe this happens like 50% of the time, and if so the host app author should set their threshold a lot higher.

Hm, maybe we need to rename `policy.crankComplete` to `deliveryComplete`, change `crankFailed` to `deliveryFailed`, and then add a `crankHappened` which means the kernel did work but not necessarily a vat.

Reply:
But, I'm ok with this PR landing without that change, if we make a ticket to improve it later.

Reply:
I think being more explicit to which kind of crank happened would indeed be good. Probably makes sense to include as part of the next step when changing the processing of the acceptance queue.